firewall.go

```diff
@@ -40,11 +41,16 @@ type conn struct {
 	Expires time.Time // Time when this conntrack entry will expire
 	Seq     uint32    // If tcp rtt tracking is enabled this will be the seq we are looking for an ack
 	Sent    time.Time // If tcp rtt tracking is enabled this will be when Seq was last set
+
+	// record why the original connection passed the firewall, so we can re-validate
+	// after ruleset changes
+	incoming  bool
+	rulesHash *string
```

Comment:
That's why I did a pointer, so it is a simple pointer equality test and not string comparison, but we still have the full rulesHash for debug logs until a simple incrementing id. Does that sway you or do you still think int would be better? I think pointer equality test will be similar performance to int equality test but I haven't verified.

Reply:
We could save space on large conntrack tables by using int32 vs 64-bit pointer. So that might be a reason.

Reply:
I don't see any real performance difference between the options, it would largely come down to memory usage. 8 bytes vs 1 byte per entry in new memory.